src/rpc/misc.cpp

```diff
@@ -1258,6 +1259,150 @@ UniValue getaddressbalance(const JSONRPCRequest& request)
     return result;
 }
 
+UniValue RanksToUniValue(CAmount lottery_anv, const Ranks& ranks, size_t total) {
+
+    UniValue rankarr(UniValue::VARR);
+    for(const auto& r : ranks) {
+        UniValue o(UniValue::VOBJ);
+
+        //percentile to two digits
+        double percentile = 
```

Comment:
Another approach might be to `sprintf("%.2f", (static_cast<double>(r.second) / static_cast<double>(total)) * 100)`

Reply:
Thanks for the code review suggestions!